pos_frequency_pipeline.py

```diff
@@ -10,16 +17,48 @@ class EmptyFileError(Exception):
 
 
 class POSFrequencyPipeline:
-    def __init__(self, corpus):
-        pass
+    def __init__(self, corpus_manager):
+        self.corpus_manager = corpus_manager
 
     def run(self):
-        pass
+        for article in self.corpus_manager.get_articles().values():
+            pos_frequencies = {}
+
+            article_path = article.get_file_path(ArtifactType.single_tagged)
+
+            with open(article_path, 'r') as text_file:
+                text = text_file.read()
+
+                if not text:
```

Comment:
and again your code is going deeper and deeper and we all remember that shallow is better than nested